### DISCUSSION TITLE
Getting the position of a drone on Qfield

### DISCUSSION THREAD

**Discussion by Ipickedausername** [opengisch/QField#4316] [Q&A]
Hello, 

Do you know if there is a way to get the position of a drone on Qfield ?
The phone is able to receive the position of the drone, so i suppose it could be displayed on the Qfield app. 

Thank you !

**Comment by VxTedxV** [comment]
I assume your phone receive its position from an app made for the drone by the drone maker ?

**Reply by Ipickedausername** [reply]
Yes exactly.

**Reply by VxTedxV** [reply]
From what I know with DJI drones for example, to access datas without the app made for theirs, the drone model must have a SDK released (code for developpers to be able to create third party apps). Actually, at least for this brand, the code is not free of use as far as I understand. So, it might be doable BUT it depends on drone brand, and drone model and maybe a different way of extraction of datas for each...
Is there anyone here who knows better and could refute what I said ?
(sorry for my bad english)

**Comment by Ipickedausername** [comment]
Alright, thank you for the information. 
This is a Mavic 3 Classic drone, from DJI. Used with DJI Fly app.

I'll search if there is a way to get the location from this app.

**Reply by fan32111** [reply]
You have to write your own app using DJI Mobile SDK and DJI UX-SDK instead of DJI Fly app.
https://developer.dji.com/mobile-sdk/
https://developer.dji.com/ux-sdk/

You get GNNS information from MSDK "class LocationCoordinate3D"
Additionally you use a NMEA Server to create messages and provide them on tcp or udp port.
eg.:https://github.com/ziroc/Nmea-Server/tree/master

Qfield can connect to NMEA Server (Settings->Positioning)

Both devices (RC Tabelt and Qfield tablet) should be in same network - connected on same hotspot
